packages/firestore/test/integration/api/query.test.ts

```diff
@@ -2061,34 +2064,133 @@ apiDescribe('Queries', (persistence: boolean) => {
       for (let i = 1; i <= 100; i++) {
         testDocs['doc' + i] = { key: i };
       }
-      return withTestCollection(persistence, testDocs, async (coll, db) => {
+
+      // The function that runs a single iteration of the test.
+      // Below this definition, there is a "while" loop that calls this
+      // function potentially multiple times.
+      const runTestIteration = async (
+        coll: CollectionReference,
+        db: Firestore
+      ): Promise<'retry' | 'passed'> => {
+        // Run a query to populate the local cache with the 100 documents
+        // and a resume token.
         const snapshot1 = await getDocs(coll);
-        expect(snapshot1.size).to.equal(100);
-        // Delete 50 docs in transaction so that it doesn't affect local cache.
+        expect(snapshot1.size, 'snapshot1.size').to.equal(100);
+
+        // Delete 50 of the 100 documents. Do this in a transaction, rather
+        // than deleteDoc(), to avoid affecting the local cache.
         await runTransaction(db, async txn => {
           for (let i = 1; i <= 50; i++) {
             txn.delete(doc(coll, 'doc' + i));
           }
         });
-        // Wait 10 seconds, during which Watch will stop tracking the query
-        // and will send an existence filter rather than "delete" events.
+
+        // Wait for 10 seconds, during which Watch will stop tracking the
+        // query and will send an existence filter rather than "delete"
+        // events when the query is resumed.
         await new Promise(resolve => setTimeout(resolve, 10000));
-        const snapshot2 = await getDocs(coll);
-        expect(snapshot2.size).to.equal(50);
-      });
+
+        // Resume the query and expect to get a snapshot with the 50
+        // remaining documents. Use some internal testing hooks to "capture"
+        // the existence filter mismatches to later verify that Watch sent a
+        // bloom filter, and it was used to avert a full requery.
+        const existenceFilterMismatches =
+          await captureExistenceFilterMismatches(async () => {
+            const snapshot2 = await getDocs(coll);
+            expect(snapshot2.size, 'snapshot2.size').to.equal(50);
+          });
+
+        // Skip the verification of the existence filter mismatch when
+        // persistence is disabled because without persistence there is no
+        // resume token specified in the subsequent call to getDocs(), and,
+        // therefore, Watch will _not_ send an existence filter.
+        if (!persistence) {
+          return 'passed';
+        }
+
+        // Skip the verification of the existence filter mismatch when
+        // testing against the Firestore emulator because the Firestore
+        // emulator does not include the `unchanged_names` bloom filter when
+        // it sends ExistenceFilter messages. Some day the emulator _may_
+        // implement this logic, at which time this short-circuit can be
+        // removed.
+        if (USE_EMULATOR) {
```

Comment:
can we remove the `(USE_EMULATOR ? it.skip : it)(` on line 2071, since we are passing the test here if it is emulator?

Reply:
Instead of skipping the entire test I made a modification later on in the test to ignore the erroneous snapshot when the query is resumed. With that, the test can run, and pass, against the emulator. It doesn't really test much, but it at least tests what it can.